juju-tutorials/how-to-fixing-attach-volume-to-a-k8s-instance.md

```diff
@@ -0,0 +1,336 @@
+
+2. The last thing is to add a new OpenStack cloud to Juju, create new credentials and create a controller. See the [OpenStack-cloud][OpenStackCloud] page for more information.
+    [note status="OpenStack metadata"]
+    It's necessary to create metadata for initializing a Juju cloud environment (creating the controller).
+    ```bash
+    juju metadata generate-image -d ~/simplestreams -i <image_id_or_name> -s <series> -r $OS_REGION_NAME -u $OS_AUTH_URL
+    ```
+    [/note]
+
+### Kubernetes
+
+Deployed charmed kubernetes on top of OpenStack with `charm-openstack-integrator` act as proxy for OpenStack and provide a control interface.
+
+```bash
+juju deploy charmed-kubernetes --overlay openstack-overlay.yaml --trust
+```
+
+Before configuration `openstack-integrator` it is necessary to source into the Openstack credential file. e.g. `source novarc`
+
+```bash
+juju config openstack-integrator auth-url=$OS_AUTH_URL region=$OS_REGION_NAME username=$OS_USERNAME user-domain-name=$OS_USER_DOMAIN_NAME password=$OS_PASSWORD project-name=$OS_PROJECT_NAME  project-domain-name=$OS_PROJECT_DOMAIN_NAME
+```
+
+### kubectl
+
+
+##### install
+
+The most common way to install `kubectl` in ubuntu is `snap`.
+
+```bash
+sudo snap install kubectl --classic
+```
+
+##### set up
+
+To configure kubectl, you must copy the configuration file from the deployed charmed kubernetes.
+
+```bash
+mkdir ~/.kube
+juju scp kubernetes-master/0:/home/ubuntu/config ~/.kube/config
+```
+[note status="VPN"]
+If your Openstack is behind a VPN, you may need to run `sshuttle` to create a tunnel for copying from the juju unit. `sshuttle -r <user@server> <juju-network>` e.g. `sshuttle -r bastion 10.5.0.0/24` (find network with `juju status -m controller`)
+[/note]
+
+### cinder volume
+
+##### create
+
+The following yaml configuration file is used to create an 8 GB cinder volume.
+
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: pvc-storage
+spec:
+  storageClassName: cdk-cinder
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 8Gi
+```
+
+The next step will be to create PersistentVolumeClaim (PVC).
+
+```bash
+kubectl apply -f <pvc-configutation-file>.yaml
+```
+
+##### attach
+
+This step shows how to attach the PVC created in the previous step. For simplicity, we will use nginx.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-nginx
+spec:
+  restartPolicy: Never
+  volumes:
+    - name: test-volume
+      persistentVolumeClaim:
+        claimName: pvc-storage
+
+  containers:
+    - name: nginx-container
+      image: nginx
+      volumeMounts:
+      - name: test-volume
+        mountPath: /usr/share/nginx/html/test_volume
+```
+
+Run the command below to deploy a nginx container. 
+
+```bash
+kubectl create -f <nginx_config>.yaml
+```
+[note status="Proxy"]
+A proxy must be set on OpenStack.
+```bash
+juju config containerd http_proxy=http://squid.internal:3128 https_proxy=http://squid.internal:3128
```

Comment:
It's a good idea to mention the possible need of proxy settings, but not every user will need one. And certainly, not the one specified here. "squid.internal" should be replaced by `<proxy-ip>`

Reply:
You are absolutely right, this is sensitive internal data.